src/components/AddItemForm.js

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import { format } from 'prettier';
+import { withFirestore } from 'react-firestore';
+import '../CSS/AddItemForm.css';
+class AddItemForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: '',
+      next_purchase: null,
+      last_purchase: null,
+      user_token: 'shpongle',
+    };
+
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleChange(event) {
+    this.setState({ name: event.target.value });
+  }
+
+  handleSubmit(event) {
+    this.props.firestore
+      .collection('itemform')
+      .add(this.state)
+      .then(function(docRef) {
+        console.log('Document written with ID: ', docRef.id);
+      })
+      .catch(function(error) {
+        console.error('Error adding document:', error);
+      });
+    event.preventDefault();
+    //To reset the input field after the user hits submit
+    this.setState({
+      name: '',
+      ext_purchase: null,
+      last_purchase: null,
+      user_token: 'shpongle',
+    });
+  }
+
+  render() {
+    return (
+      <form onSubmit={this.handleSubmit}>
+        <label class="name">
+          Name:
+          <input
+            type="text"
+            value={this.state.name}
+            onChange={this.handleChange}
+          />
+          <input type="submit" value="Submit" />
```

Comment:
The submit button should be at the end of the form. You have a text field then the submit button, followed by two more form inputs. That is not good for accessibility.